gdal-3.2.1/man/man1/gdalinfo.1
.\" Man page generated from reStructuredText.
.
.TH "GDALINFO" "1" "Dec 29, 2020" "" "GDAL"
.SH NAME
gdalinfo \- Lists various information about a GDAL supported raster dataset
.
.nr rst2man-indent-level 0
.
.de1 rstReportMargin
\\$1 \\n[an-margin]
level \\n[rst2man-indent-level]
level margin: \\n[rst2man-indent\\n[rst2man-indent-level]]
-
\\n[rst2man-indent0]
\\n[rst2man-indent1]
\\n[rst2man-indent2]
..
.de1 INDENT
.\" .rstReportMargin pre:
. RS \\$1
. nr rst2man-indent\\n[rst2man-indent-level] \\n[an-margin]
. nr rst2man-indent-level +1
.\" .rstReportMargin post:
..
.de UNINDENT
. RE
.\" indent \\n[an-margin]
.\" old: \\n[rst2man-indent\\n[rst2man-indent-level]]
.nr rst2man-indent-level -1
.\" new: \\n[rst2man-indent\\n[rst2man-indent-level]]
.in \\n[rst2man-indent\\n[rst2man-indent-level]]u
..
.SH SYNOPSIS
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
gdalinfo [\-\-help\-general] [\-json] [\-mm] [\-stats] [\-hist] [\-nogcp] [\-nomd]
         [\-norat] [\-noct] [\-nofl] [\-checksum] [\-proj4]
         [\-listmdd] [\-mdd domain|\(gaall\(ga]* [\-wkt_format WKT1|WKT2|...]
         [\-sd subdataset] [\-oo NAME=VALUE]* [\-if format]* datasetname
.ft P
.fi
.UNINDENT
.UNINDENT
.SH DESCRIPTION
.sp
\fBgdalinfo\fP program lists various information about a GDAL supported
raster dataset.
.sp
The following command line parameters can appear in any order
.INDENT 0.0
.TP
.B \-json
Display the output in json format.
.UNINDENT
.INDENT 0.0
.TP
.B \-mm
Force computation of the actual min/max values for each band in the
dataset.
.UNINDENT
.INDENT 0.0
.TP
.B \-stats
Read and display image statistics. Force computation if no
statistics are stored in an image.
.UNINDENT
.INDENT 0.0
.TP
.B \-approx_stats
Read and display image statistics. Force computation if no
statistics are stored in an image. However, they may be computed
based on overviews or a subset of all tiles. Useful if you are in a
hurry and don\(aqt want precise stats.
.UNINDENT
.INDENT 0.0
.TP
.B \-hist
Report histogram information for all bands.
.UNINDENT
.INDENT 0.0
.TP
.B \-nogcp
Suppress ground control points list printing. It may be useful for
datasets with huge amount of GCPs, such as L1B AVHRR or HDF4 MODIS
which contain thousands of them.
.UNINDENT
.INDENT 0.0
.TP
.B \-nomd
Suppress metadata printing. Some datasets may contain a lot of
metadata strings.
.UNINDENT
.INDENT 0.0
.TP
.B \-norat
Suppress printing of raster attribute table.
.UNINDENT
.INDENT 0.0
.TP
.B \-noct
Suppress printing of color table.
.UNINDENT
.INDENT 0.0
.TP
.B \-checksum
Force computation of the checksum for each band in the dataset.
.UNINDENT
.INDENT 0.0
.TP
.B \-listmdd
List all metadata domains available for the dataset.
.UNINDENT
.INDENT 0.0
.TP
.B \-mdd <domain>|all
adds metadata using:
.sp
\fBdomain\fP Report metadata for the specified domain.
.sp
\fBall\fP Report metadata for all domains.
.UNINDENT
.INDENT 0.0
.TP
.B \-nofl
Only display the first file of the file list.
.UNINDENT
.INDENT 0.0
.TP
.B \-wkt_format WKT1|WKT2|WKT2_2015|WKT2_2018
WKT format used to display the SRS.
Currently the supported values are:
.sp
\fBWKT1\fP
.sp
\fBWKT2\fP (latest WKT version, currently \fIWKT2_2018\fP)
.sp
\fBWKT2_2015\fP
.sp
\fBWKT2_2018\fP
.sp
New in version 3.0.0.

.UNINDENT
.INDENT 0.0
.TP
.B \-sd <n>
If the input dataset contains several subdatasets read and display
a subdataset with specified \fBn\fP number (starting from 1).
This is an alternative of giving the full subdataset name.
.UNINDENT
.INDENT 0.0
.TP
.B \-proj4
Report a PROJ.4 string corresponding to the file\(aqs coordinate system.
.UNINDENT
.INDENT 0.0
.TP
.B \-oo <NAME=VALUE>
Dataset open option (format specific).
.UNINDENT
.INDENT 0.0
.TP
.B \-if <format>
Format/driver name to be attempted to open the input file(s). It is generally
not necessary to specify it, but it can be used to skip automatic driver
detection, when it fails to select the appropriate driver.
This option can be repeated several times to specify several candidate drivers.
.sp
New in version 3.2.

.UNINDENT
.sp
The gdalinfo will report all of the following (if known):
.INDENT 0.0
.IP \(bu 2
The format driver used to access the file.
.IP \(bu 2
Raster size (in pixels and lines).
.IP \(bu 2
The coordinate system for the file (in OGC WKT).
.IP \(bu 2
The geotransform associated with the file (rotational coefficients
are currently not reported).
.IP \(bu 2
Corner coordinates in georeferenced, and if possible lat/long based
on the full geotransform (but not GCPs).
.IP \(bu 2
Ground control points.
.IP \(bu 2
File wide (including subdatasets) metadata.
.IP \(bu 2
Band data types.
.IP \(bu 2
Band color interpretations.
.IP \(bu 2
Band block size.
.IP \(bu 2
Band descriptions.
.IP \(bu 2
Band min/max values (internally known and possibly computed).
.IP \(bu 2
Band checksum (if computation asked).
.IP \(bu 2
Band NODATA value.
.IP \(bu 2
Band overview resolutions available.
.IP \(bu 2
Band unit type (i.e.. "meters" or "feet" for elevation bands).
.IP \(bu 2
Band pseudo\-color tables.
.UNINDENT
.SH C API
.sp
This utility is also callable from C with \fBGDALInfo()\fP\&.
.sp
New in version 2.1.

.SH EXAMPLE
.INDENT 0.0
.INDENT 3.5
.sp
.nf
.ft C
gdalinfo ~/openev/utm.tif
Driver: GTiff/GeoTIFF
Size is 512, 512
Coordinate System is:
PROJCS["NAD27 / UTM zone 11N",
    GEOGCS["NAD27",
        DATUM["North_American_Datum_1927",
            SPHEROID["Clarke 1866",6378206.4,294.978698213901]],
        PRIMEM["Greenwich",0],
        UNIT["degree",0.0174532925199433]],
    PROJECTION["Transverse_Mercator"],
    PARAMETER["latitude_of_origin",0],
    PARAMETER["central_meridian",\-117],
    PARAMETER["scale_factor",0.9996],
    PARAMETER["false_easting",500000],
    PARAMETER["false_northing",0],
    UNIT["metre",1]]
Origin = (440720.000000,3751320.000000)
Pixel Size = (60.000000,\-60.000000)
Corner Coordinates:
Upper Left  (  440720.000, 3751320.000) (117d38\(aq28.21"W, 33d54\(aq8.47"N)
Lower Left  (  440720.000, 3720600.000) (117d38\(aq20.79"W, 33d37\(aq31.04"N)
Upper Right (  471440.000, 3751320.000) (117d18\(aq32.07"W, 33d54\(aq13.08"N)
Lower Right (  471440.000, 3720600.000) (117d18\(aq28.50"W, 33d37\(aq35.61"N)
Center      (  456080.000, 3735960.000) (117d28\(aq27.39"W, 33d45\(aq52.46"N)
Band 1 Block=512x16 Type=Byte, ColorInterp=Gray
.ft P
.fi
.UNINDENT
.UNINDENT
.SH AUTHOR
Frank Warmerdam <warmerdam@pobox.com>
.SH COPYRIGHT
1998-2020
.\" Generated by docutils manpage writer.
.


gdal-3.0.1/swig/csharp/apps/GDALInfo.cs
/******************************************************************************
 * $Id: GDALInfo.cs f070adf64950cae1c6cc86b104ba835c29df06b1 2016-08-28 06:06:11Z Kurt Schwehr $
 *
 * Name:     GDALInfo.cs
 * Project:  GDAL CSharp Interface
 * Purpose:  A sample app to read GDAL raster data information.
 * Author:   Tamas Szekeres, szekerest@gmail.com
 *
 ******************************************************************************
 * Copyright (c) 2007, Tamas Szekeres
 *
 * Permission is hereby granted, free of charge, to any person obtaining a
 * copy of this software and associated documentation files (the "Software"),
 * to deal in the Software without restriction, including without limitation
 * the rights to use, copy, modify, merge, publish, distribute, sublicense,
 * and/or sell copies of the Software, and to permit persons to whom the
 * Software is furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
 * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
 * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
 * DEALINGS IN THE SOFTWARE.
 *****************************************************************************/

using System;

using OSGeo.GDAL;
using OSGeo.OSR;


/**

 * <p>Title: GDAL C# GDALRead example.</p>
 * <p>Description: A sample app to read GDAL raster data information.</p>
 * @author Tamas Szekeres (szekerest@gmail.com)
 * @version 1.0
 */



/// <summary>
/// A C# based sample to read GDAL raster data information.
/// </summary>

class GDALInfo {

	public static void usage()

	{
		Console.WriteLine("usage: gdalinfo {GDAL dataset name}");
		System.Environment.Exit(-1);
	}

    public static void Main(string[] args)
    {

        if (args.Length != 1) usage();

        Console.WriteLine("");

        try
        {
            /* -------------------------------------------------------------------- */
            /*      Register driver(s).                                             */
            /* -------------------------------------------------------------------- */
            Gdal.AllRegister();

            /* -------------------------------------------------------------------- */
            /*      Open dataset.                                                   */
            /* -------------------------------------------------------------------- */
            Dataset ds = Gdal.Open( args[0], Access.GA_ReadOnly );

            if (ds == null)
            {
                Console.WriteLine("Can't open " + args[0]);
                System.Environment.Exit(-1);
            }

            Console.WriteLine("Raster dataset parameters:");
            Console.WriteLine("  Projection: " + ds.GetProjectionRef());
            Console.WriteLine("  RasterCount: " + ds.RasterCount);
            Console.WriteLine("  RasterSize (" + ds.RasterXSize + "," + ds.RasterYSize + ")");

            /* -------------------------------------------------------------------- */
            /*      Get driver                                                      */
            /* -------------------------------------------------------------------- */
            Driver drv = ds.GetDriver();

            if (drv == null)
            {
                Console.WriteLine("Can't get driver.");
                System.Environment.Exit(-1);
            }

            Console.WriteLine("Using driver " + drv.LongName);

            /* -------------------------------------------------------------------- */
            /*      Get metadata                                                    */
            /* -------------------------------------------------------------------- */
            string[] metadata = ds.GetMetadata("");
            if (metadata.Length > 0)
            {
                Console.WriteLine("  Metadata:");
                for (int iMeta = 0; iMeta < metadata.Length; iMeta++)
                {
                    Console.WriteLine("    " + iMeta + ":  " + metadata[iMeta]);
                }
                Console.WriteLine("");
            }

            /* -------------------------------------------------------------------- */
            /*      Report "IMAGE_STRUCTURE" metadata.                              */
            /* -------------------------------------------------------------------- */
            metadata = ds.GetMetadata("IMAGE_STRUCTURE");
            if (metadata.Length > 0)
            {
                Console.WriteLine("  Image Structure Metadata:");
                for (int iMeta = 0; iMeta < metadata.Length; iMeta++)
                {
                    Console.WriteLine("    " + iMeta + ":  " + metadata[iMeta]);
                }
                Console.WriteLine("");
            }

            /* -------------------------------------------------------------------- */
            /*      Report subdatasets.                                             */
            /* -------------------------------------------------------------------- */
            metadata = ds.GetMetadata("SUBDATASETS");
            if (metadata.Length > 0)
            {
                Console.WriteLine("  Subdatasets:");
                for (int iMeta = 0; iMeta < metadata.Length; iMeta++)
                {
                    Console.WriteLine("    " + iMeta + ":  " + metadata[iMeta]);
                }
                Console.WriteLine("");
            }

            /* -------------------------------------------------------------------- */
            /*      Report geolocation.                                             */
            /* -------------------------------------------------------------------- */
            metadata = ds.GetMetadata("GEOLOCATION");
            if (metadata.Length > 0)
            {
                Console.WriteLine("  Geolocation:");
                for (int iMeta = 0; iMeta < metadata.Length; iMeta++)
                {
                    Console.WriteLine("    " + iMeta + ":  " + metadata[iMeta]);
                }
                Console.WriteLine("");
            }

            /* -------------------------------------------------------------------- */
            /*      Report corners.                                                 */
            /* -------------------------------------------------------------------- */
            Console.WriteLine( "Corner Coordinates:" );
            Console.WriteLine("  Upper Left (" + GDALInfoGetPosition( ds, 0.0, 0.0) + ")");
            Console.WriteLine("  Lower Left (" + GDALInfoGetPosition( ds, 0.0, ds.RasterYSize) + ")");
            Console.WriteLine("  Upper Right (" + GDALInfoGetPosition( ds, ds.RasterXSize, 0.0) + ")");
            Console.WriteLine("  Lower Right (" + GDALInfoGetPosition( ds, ds.RasterXSize, ds.RasterYSize) + ")");
            Console.WriteLine("  Center (" + GDALInfoGetPosition( ds, ds.RasterXSize / 2, ds.RasterYSize / 2) + ")");
            Console.WriteLine("");

            /* -------------------------------------------------------------------- */
            /*      Report projection.                                              */
            /* -------------------------------------------------------------------- */
            string projection = ds.GetProjectionRef();
            if (projection != null)
            {
                SpatialReference srs = new SpatialReference(null);
                if (srs.ImportFromWkt(ref projection) == 0)
                {
                    string wkt;
                    srs.ExportToPrettyWkt(out wkt, 0);
                    Console.WriteLine("Coordinate System is:");
                    Console.WriteLine(wkt);
                }
                else
                {
                    Console.WriteLine("Coordinate System is:");
                    Console.WriteLine(projection);
                }
            }

            /* -------------------------------------------------------------------- */
            /*      Report GCPs.                                                    */
            /* -------------------------------------------------------------------- */
            if( ds.GetGCPCount( ) > 0 )
            {
                Console.WriteLine( "GCP Projection: ", ds.GetGCPProjection());
                GCP[] GCPs = ds.GetGCPs();
                for( int i = 0; i < ds.GetGCPCount(); i++ )
                {
                    Console.WriteLine("GCP[" + i + "]: Id=" + GCPs[i].Id + ", Info=" + GCPs[i].Info);
                    Console.WriteLine("          (" + GCPs[i].GCPPixel + "," + GCPs[i].GCPLine + ") -> ("
                                + GCPs[i].GCPX + "," + GCPs[i].GCPY + "," + GCPs[i].GCPZ + ")");
                    Console.WriteLine("");
                }
                Console.WriteLine("");

                double[] transform = new double[6];
                Gdal.GCPsToGeoTransform(GCPs, transform, 0);
                Console.WriteLine("GCP Equivalent geotransformation parameters: ", ds.GetGCPProjection());
                for (int i = 0; i < 6; i++)
                    Console.WriteLine("t[" + i + "] = " + transform[i].ToString());
                Console.WriteLine("");
            }

            /* -------------------------------------------------------------------- */
            /*      Get raster band                                                 */
            /* -------------------------------------------------------------------- */
            for (int iBand = 1; iBand <= ds.RasterCount; iBand++)
            {
                Band band = ds.GetRasterBand(iBand);
                Console.WriteLine("Band " + iBand + " :");
                Console.WriteLine("   DataType: " + Gdal.GetDataTypeName(band.DataType));
                Console.WriteLine("   ColorInterpretation: " + Gdal.GetColorInterpretationName(band.GetRasterColorInterpretation()));
                ColorTable ct = band.GetRasterColorTable();
				if (ct != null)
					Console.WriteLine("   Band has a color table with " + ct.GetCount() + " entries.");

				Console.WriteLine("   Description: " + band.GetDescription());
                Console.WriteLine("   Size (" + band.XSize + "," + band.YSize + ")");
                int BlockXSize, BlockYSize;
                band.GetBlockSize(out BlockXSize, out BlockYSize);
                Console.WriteLine("   BlockSize (" + BlockXSize + "," + BlockYSize + ")");
                double val;
                int hasval;
                band.GetMinimum(out val, out hasval);
                if (hasval != 0) Console.WriteLine("   Minimum: " + val.ToString());
                band.GetMaximum(out val, out hasval);
                if (hasval != 0) Console.WriteLine("   Maximum: " + val.ToString());
                band.GetNoDataValue(out val, out hasval);
                if (hasval != 0) Console.WriteLine("   NoDataValue: " + val.ToString());
                band.GetOffset(out val, out hasval);
                if (hasval != 0) Console.WriteLine("   Offset: " + val.ToString());
                band.GetScale(out val, out hasval);
                if (hasval != 0) Console.WriteLine("   Scale: " + val.ToString());

                for (int iOver = 0; iOver < band.GetOverviewCount(); iOver++)
                {
                    Band over = band.GetOverview(iOver);
                    Console.WriteLine("      OverView " + iOver + " :");
                    Console.WriteLine("         DataType: " + over.DataType);
                    Console.WriteLine("         Size (" + over.XSize + "," + over.YSize + ")");
                    Console.WriteLine("         PaletteInterp: " + over.GetRasterColorInterpretation().ToString());
                }
            }
        }
        catch (Exception e)
        {
            Console.WriteLine("Application error: " + e.Message);
        }
    }

    private static string GDALInfoGetPosition(Dataset ds, double x, double y)
    {
        double[] adfGeoTransform = new double[6];
        double	dfGeoX, dfGeoY;
        ds.GetGeoTransform(adfGeoTransform);

        dfGeoX = adfGeoTransform[0] + adfGeoTransform[1] * x + adfGeoTransform[2] * y;
        dfGeoY = adfGeoTransform[3] + adfGeoTransform[4] * x + adfGeoTransform[5] * y;

        return dfGeoX.ToString() + ", " + dfGeoY.ToString();
    }
}